### DISCUSSION TITLE
Rewrite E2 Arrays

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2872] [Suggestions]
Make arrays more like E2 tables because I'm sick of them.

- `class E2Array`
  - `[integer] any` - Sequential Data
  - `string? Type` - Restricted type or nil
  - `integer? SizeL` - Size limit
  - `string[]? TypeIds` - Defined as the array of typeids if `Type` is undefined

And then copy the getters/setters in #2817

Arrays don't use string indices so using those shouldn't change anything. `SizeL` is for future consideration (`Type` can still be used as an optimization trick).

Biggest issue may be serializing, especially deserializing from old arrays. Need to see how von handles this.

Not even suggesting this for arrays/tables inside arrays, just want them to be easier to use.

**Comment by vurvdev** [comment]
This isn't gonna happen. Ship has long sailed since arrays could be changed. Doing this would just severely complicate things. In the future a statically typed array should exist, not this which is just another table.
